ray-curator/ray_curator/stages/modules/add_id.py

```diff
@@ -0,0 +1,51 @@
+from dataclasses import dataclass
+
+from ray_curator.stages.base import ProcessingStage
+from ray_curator.tasks import DocumentBatch
+
+
+@dataclass
+class AddId(ProcessingStage[DocumentBatch, DocumentBatch]):
+    """
+    The module responsible for adding unique IDs to each document record.
+
+    This stage adds a unique identifier to each document in the batch by combining
+    the task UUID with a sequential index.
+
+    Args:
+        id_field (str): The field where the generated ID will be stored.
+    """
+
+    id_field: str
```

Comment:
Could we also support adding a user specified prefix to the ID. It's fairly common for folks to prefix with dataset-name or some other naming scheme. The id could be in the format `{user-prefix}_{uuid}_{i}` for each row

Reply:
Added